src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2323,9 +2323,8 @@ NetworkOPsImp::getServerInfo(bool human, bool admin, bool counters)
     if (!app_.config().SERVER_DOMAIN.empty())
         info[jss::server_domain] = app_.config().SERVER_DOMAIN;
 
-    if (!app_.config().reporting())
-        if (auto const netid = app_.overlay().networkID())
-            info[jss::network_id] = static_cast<Json::UInt>(*netid);
+    if (auto const netid = app_.overlay().networkID(); netid.has_value())
```

Comment:
I don't know if removing the check for reporting mode is correct. Maybe @cjcobb23 can chime in?

Second, the `if` is unnecessarily complicated. It should be reverted to its previous form:

```
if (auto const netid = app_.overlay().networkID())
    info[jss::network_id] = static_cast<Json::UInt>(*netid);
```

Reply:
Okay, I didn't want to run into undefined behavior, so I did explicit checks. I'll revert it 👍